src/smc-webapp/editor.coffee

```diff
@@ -1932,13 +1932,33 @@ class CodeMirrorEditor extends FileEditor
         for i, cm of [@codemirror, @codemirror1]
             size = @local_storage("font_size#{i}")
             if size?
-                @set_font_size(cm, size)
+                if size < 2
+                    size = 2
+                @set_font_size(cm, size, 0)
             else if @default_font_size?
-                @set_font_size(cm, @default_font_size)
+                @set_font_size(cm, @default_font_size, 0)
 
-    set_font_size: (cm, size) =>
+    set_font_size: (cm, size, delta) =>
         if size > 1
             elt = $(cm.getWrapperElement())
+            if delta != 0
+                for html, i in elt.find('.sagews-output-html').children()
+                    original_font_size = parseInt($(html).css('font-size'))
+                    $(html).css('font-size', original_font_size + delta + 'px')
+                    @local_storage("html_font#{i + html}", original_font_size + delta)
```

Comment:
uhm, and all that into local storage? does it even support keys of arbitrary length? or do I miss what `html` is? I think I should finish reading this first ...